upload-chunked-file.js

```diff
@@ -0,0 +1,43 @@
+function chunkFileUpload(file, serverUrl, chunkSize) {
+    const totalChunks = Math.ceil(file.size / chunkSize);
+    let chunkIndex = 0;
+
+    (function uploadNextChunk() {
+        const start = chunkIndex * chunkSize;
+        const end = Math.min(start + chunkSize, file.size);
+        const chunk = file.slice(start, end);
+
```

Comment:
Consider using `let` or `const` instead of `var` to declare `xhr`. This improves the readability and maintainability of your code by avoiding unexpected scope-related issues.